tardis/widgets/base.py

```diff
@@ -0,0 +1,55 @@
+from tardis.base import run_tardis
+from tardis.io.atom_data.util import download_atom_data
+from tardis.util.base import atomic_number2element_symbol, species_tuple_to_string
+
+import pandas as pd
+import numpy as np
+import qgrid
+import ipywidgets as ipw
+
+# TODO: Fetch saved Simulation object from memory
+sim = run_tardis('tardis_example.yml')  # TODO: Make the file path work
+
+class ShellInfo():
+    def __init__(self, sim_model=sim):
+        self.sim_model = sim_model
+
+    def shells_data(self):
+        shells_temp_w = pd.DataFrame({'Rad. Temp.': self.sim_model.model.t_rad,
+                                'W': self.sim_model.model.w},
+                                index=range(1, 21))
```

Comment:
Not every model will have 20 shells. Let's generalize 21 to len(self.sim_model.model.t_rad)+1

Reply:
good catch